testsuite/test_interface.py

```diff
@@ -35,11 +43,30 @@ def test_submission(client):
     submission = Submission.objects.all()[0]
 
     assert submission.state == submission.STATE_NEW
-    assert submission.assignment.code == 'pc-a0'
+    assert submission.assignment.code == 'a0'
     assert submission.archive_size > 0
     assert submission.vmck_job_id > 0
 
+    count = 0
     while submission.state != submission.STATE_DONE:
-        time.sleep(2)
+        submission = Submission.objects.all()[0]
 
-    assert submission.score > 0
+        time.sleep(10)
+
+        count += 1
+        if count == 18:
```

Comment:
I suggest you use [`time()`](https://docs.python.org/3/library/time.html#time.time) to decide when to bail from the loop, it's easier to understand when reading the code.